redash/handlers/embed.py

```diff
@@ -125,7 +125,12 @@ def embed(query_id, visualization_id, org_slug=None):
 def public_dashboard(token, org_slug=None):
     # TODO: verify object is a dashboard?
     if not isinstance(current_user, models.ApiUser):
-        api_key = models.ApiKey.get_by_api_key(token)
```

Comment:
You can use the [`get_object_or_404`](https://github.com/getredash/redash/blob/master/redash/handlers/base.py#L68-L72) helper method: 

`api_key = get_object_or_404(models.ApiKey.get_by_api_key, token)`.